scalalib/src/main/scala/MacroLib.scala

```diff
@@ -102,39 +102,168 @@ case class SRAMMacro(
   override def typeStr = "sram"
 }
 object SRAMMacro {
-  def parseJSON(json:Map[String, JsValue]): Option[SRAMMacro] = {
+  def parseJSON(json: Map[String, JsValue]): Option[SRAMMacro] = {
     val name: String = json.get("name") match {
-      case Some(x:JsString) => x.as[String]
+      case Some(x: JsString) => x.as[String]
       case _ => return None
     }
     val width: Int = json.get("width") match {
-      case Some(x:JsNumber) => x.value.intValue
+      case Some(x: JsNumber) => x.value.intValue
       case _ => return None
     }
     val depth: Int = json.get("depth") match {
-      case Some(x:JsNumber) => x.value.intValue
+      case Some(x: JsNumber) => x.value.intValue
       case _ => return None
     }
     val family: String = json.get("family") match {
-      case Some(x:JsString) => x.as[String]
+      case Some(x: JsString) => x.as[String]
       case _ => "" // optional
     }
     val ports: Seq[MacroPort] = json.get("ports") match {
-      case Some(x:JsArray) => x.as[List[Map[String, JsValue]]] map { a => { val b = MacroPort.parseJSON(a, width, depth); if (b == None) { return None } else b.get } }
+      case Some(x: JsArray) => x.as[List[Map[String, JsValue]]] map { a =>
+        val b = MacroPort.parseJSON(a, width, depth); if (b == None) {
+          return None
+        } else b.get
+      }
       case _ => List()
     }
     if (ports.length == 0) {
       // Can't have portless memories.
       return None
     }
     val extraPorts: Seq[MacroExtraPort] = json.get("extra ports") match {
-      case Some(x:JsArray) => x.as[List[Map[String, JsValue]]] map { a => { val b = MacroExtraPort.parseJSON(a); if (b == None) { return None } else b.get } }
+      case Some(x: JsArray) => x.as[List[Map[String, JsValue]]] map { a =>
+        val b = MacroExtraPort.parseJSON(a); if (b == None) {
+          return None
+        } else b.get
+      }
       case _ => List()
     }
     Some(SRAMMacro(name, width, depth, family, ports, extraPorts))
   }
 }
 
+// SRAM compiler
+case class SRAMGroup(
+  name: Seq[String],
+  family: String,
+  vt: Seq[String],
+  mux: Int,
+  depth: Range,
```

Comment:
Similar concern as in the other PR

Reply:
Ok, after talking to Colin, we will make it inclusive on the maximum value.